React/Views/RCTMapManager.m

```diff
@@ -90,14 +150,20 @@ - (MKAnnotationView *)mapView:(__unused MKMapView *)mapView viewForAnnotation:(R
   annotationView.canShowCallout = true;
   annotationView.animatesDrop = annotation.animateDrop;
 
-  annotationView.leftCalloutAccessoryView = nil;
-  if (annotation.hasLeftCallout) {
-    annotationView.leftCalloutAccessoryView = [UIButton buttonWithType:UIButtonTypeDetailDisclosure];
-  }
+  for (NSString *side in @[@"left", @"right"]) {
+    NSString *accessoryViewSelector = [NSString stringWithFormat:@"%@CalloutAccessoryView", side];
+    NSString *typeSelector = [NSString stringWithFormat:@"%@CalloutType", side];
+    NSString *hasCheckSelector = [NSString stringWithFormat:@"has%@Callout", [side capitalizedString]];
+    NSString *configSelector = [NSString stringWithFormat:@"%@CalloutConfig", side];
 
-  annotationView.rightCalloutAccessoryView = nil;
-  if (annotation.hasRightCallout) {
-    annotationView.rightCalloutAccessoryView = [UIButton buttonWithType:UIButtonTypeDetailDisclosure];
+    [annotationView setValue:nil forKey:accessoryViewSelector];
+    if ([annotation valueForKey:hasCheckSelector]) {
+      [annotationView setValue:[UIButton buttonWithType:UIButtonTypeDetailDisclosure] forKey:accessoryViewSelector];
+
+      if ([[annotation valueForKey:typeSelector] integerValue] == RCTPointAnnotationTypeImage) {
+        [annotationView setValue:[self generateCalloutImageAccessory:[annotation valueForKey:configSelector]] forKey:accessoryViewSelector];
+      }
+    }
```

Comment:
@machard good point and thanks for the comment. But since this PR is 'on hold', I am not sure whether it's worth changing the code now or just wait for react components in callouts